Source/BugsnagThread.m

```diff
@@ -7,7 +7,103 @@
 //
 
 #import "BugsnagThread.h"
+#import "BugsnagCollections.h"
+#import "BugsnagStacktrace.h"
+#import "BugsnagKeys.h"
+
+@interface BugsnagStacktrace ()
+- (NSArray *)toArray;
+@end
+
+@interface BugsnagThread ()
+@property BugsnagStacktrace *trace;
+@end
+
+@interface BugsnagStacktrace ()
+@property NSMutableArray<BugsnagStackframe *> *trace;
+@end
 
 @implementation BugsnagThread
 
+- (instancetype)initWithThread:(NSDictionary *)thread binaryImages:(NSArray *)binaryImages {
+    if (self = [super init]) {
+        self.errorReportingThread = [thread[@"crashed"] boolValue];
+        self.id = [thread[@"index"] stringValue];
+        self.type = BSGThreadTypeCocoa;
+
+        NSArray *backtrace = thread[@"backtrace"][@"contents"];
+        self.trace = [[BugsnagStacktrace alloc] initWithTrace:backtrace binaryImages:binaryImages];
+    }
+    return self;
+}
+
+- (NSMutableArray<BugsnagStackframe *> *)stacktrace {
+    return self.trace.trace;
+}
+
+- (NSDictionary *)toDict {
+    NSMutableDictionary *dict = [NSMutableDictionary new];
+    BSGDictInsertIfNotNil(dict, self.id, @"id");
+    BSGDictInsertIfNotNil(dict, self.name, @"name");
+    BSGDictSetSafeObject(dict, @(self.errorReportingThread), @"errorReportingThread");
+    BSGDictSetSafeObject(dict, self.type == BSGThreadTypeCocoa ? @"cocoa" : @"reactnativejs", @"type");
+    BSGDictSetSafeObject(dict, @(self.errorReportingThread), @"errorReportingThread");
+    BSGDictSetSafeObject(dict, [self.trace toArray], @"stacktrace");
+    return dict;
+}
+
++ (NSMutableArray *)serializeThreads:(NSMutableDictionary *)event threads:(NSArray<BugsnagThread *> *)threads {
```

Comment:
Is `event` used?  This method is only called from `BugsnagEvent` so the parameter could be left off.

Reply:
It doesn't seem to be used - I'll remove the parameter.